From f68b2363f94863dbc7267a08b708753fb1cd5211 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Tue, 16 Mar 2021 20:49:20 +0100
Subject: [PATCH] Work around a regression in \`windres\`

It seems that `windres` used to accept a partial command-line (i.e a
program _and_ some command-line parameters) as `--preprocessor`
argument, but now it insists on taking the program _without_ any
parameters.

Let's work around that.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 src/Makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index fb7e8cf..b906a1c 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -157,7 +157,9 @@ $(BINDIR)/%.o: %.c
 	$(CC) -c $(DEPOPT) $(CCFLAGS) $(CPPFLAGS) -fstack-check $< -o $(BINDIR)/$*.o
 
 $(BINDIR)/%.o: %.rc %.h %.mft
-	$(RC) -c 65001 --preprocessor '$(CC) -E -xc -DRC_INVOKED $(DEPOPT) $(CPPFLAGS)' $< $(BINDIR)/$*.o
+	printf >preprocessor.sh '#!/bin/sh\n\n$(CC) -E -xc -DRC_INVOKED $(DEPOPT) $(CPPFLAGS) "$$@"\n'
+	chmod +x preprocessor.sh
+	$(RC) -c 65001 --preprocessor "$$PWD/preprocessor.sh" $< $(BINDIR)/$*.o
 	-mv $*.d $(BINDIR)/
 
 exe := $(NAME).exe
-- 
2.30.2

